src/main/java/duke2/Parser.java

```diff
@@ -0,0 +1,228 @@
+package duke2;// package duke1;
+
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+
+
+public class Parser {
+
+    /**
+     * Writes given text to file in given path
+     * 
+     * @param path of file
+     * @param text to add
+     * @throws IOException
+     * @returns void
+     */
+    public static void writeToFile(String path, String text) throws IOException {
+        FileWriter fw = new FileWriter(path, true);
+        fw.write(text);
+        fw.write(System.getProperty("line.separator"));
+        fw.close();
+
+    }
+
+    /**
+     * Formats date to mmm-dd-yyyy
+     * 
+     * @param date to format
+     * @return formatted date in mmm-dd-yyy
+     */
+    public static String dateFormat(String date) {
+        LocalDate d1 = LocalDate.parse(date);
+        return d1.format(DateTimeFormatter.ofPattern("MMM dd yyyy "));
+    }
+
+    /**
+     * Formats time to 12h
+     * 
+     * @param time to format
+     * @return formatted time in 12h
+     */
+    public static String timeFormat(String time) {
+        int i = Integer.parseInt(time);
+        if (i > 0000 && i < 1100) {
+            return i + "am";
+        } else if (i >= 1300 && i < 2359) {
+            char[] c = time.toCharArray();
+            String min = new StringBuilder().append(c[2]).append(c[3]).toString();
+            String hour = new StringBuilder().append(c[0]).append(c[1]).toString();
+
+            int h = Integer.parseInt(hour) - 12;
+            return String.valueOf(h) + min + "pm";
+        } else {
+            return i + "pm";
+        }
+
+    }
+
+    /**
+     * Parses string to create new task object
+     * 
+     * @param task to create
+     * @return new task from string
+     */
+    public static Task parseStringToTask(String task) {
```

Comment:
Your `parseStringToTask` method is really long and has a lot of nested if-else checks :<
Consider refactoring it!